frontend/src/metabase/parameters/utils/dashboards.ts

```diff
@@ -118,39 +180,41 @@ export function getMappingsByParameter(metadata, dashboard) {
 }
 
 export function getDashboardParametersWithFieldMetadata(
-  metadata,
-  dashboard,
-  mappingsByParameter,
-) {
+  metadata: Metadata,
+  dashboard: Dashboard,
+  mappingsByParameter: NestedMappingsMap,
+): ExtendedFieldFilterUiParameter[] {
   return ((dashboard && dashboard.parameters) || []).map(parameter => {
-    const mappings = _.flatten(
+    const mappings: ExtendedMapping[] = _.flatten(
       _.map(mappingsByParameter[parameter.id] || {}, _.values),
     );
 
     // we change out widgets if a parameter is connected to non-field targets
     const hasOnlyFieldTargets = mappings.every(x => x.field_id != null);
 
-    const fields = _.uniq(
+    const fields: Field[] = _.uniq(
       mappings
         .map(mapping => mapping.field)
-        .filter(field => field != null)
+        .filter((field): field is Field => field != null)
         .map(field => field.target ?? field),
       field => field.id,
     );
 
     // get the unique list of field IDs these mappings reference
-    const fieldIds = _.chain(mappings)
-      .map(m => m.field_id)
-      .uniq()
-      .filter(fieldId => fieldId != null)
-      .value();
-
-    const fieldIdsWithFKResolved = _.chain(fieldIds)
-      .map(id => metadata.field(id))
-      .filter(f => f)
-      .map(f => (f.target || f).id)
-      .uniq()
-      .value();
+    const fieldIds: (string | number)[] = _.uniq(
```

Comment:
Had to remove `_.chain` here and for `fieldIdsWithFKResolved` so that TS could infer types.